### DISCUSSION TITLE
Same code, multiple panels

### DISCUSSION THREAD

**Discussion by psheehymn** [gapitio/gapit-htmlgraphics-panel#169] [Q&A]
I have a dashboard that repeats the same HTML Graphics panel multiple times (15 so far) to display status for a variety of different topic areas within a portal (the queries are the only variant in each panel). The JavaScript in the panel is rather extensive.

Whenever I need to tweak the code, I then have to propagate the new version to all the panels. While not a difficult task, it's an onerous one and introduces the potential for a defect if I miss updating one or more panels due to a distraction or interruption.

Is there any way to put the JS code in a central location and have all the panels use the same copy, so updating it once automatically updates all the panels? I looked thru the documentation and couldn't find any reference to this, but I may not have been searching with the right keywords. Any pointers or ideas would be appreciated.

**Comment by ZuperZee** [comment]
You could do it through the Grafana API, but it requires some time to make it work how you explained.

The watcher part of htmlgraphics svg bundler uses the API call to update the grafana dashboard <https://github.com/gapitio/htmlgraphics-svg-bundler-template/blob/67279d36b9fc14f31765b2f2e8fe7dfd1fd0a852/scripts/watch.js#L120>.

It would probably be easier to update the panels manually than to create the script.
